pmd-cpp/src/main/java/net/sourceforge/pmd/lang/cpp/CppCharStream.java

```diff
@@ -59,4 +65,17 @@ public String GetImage() {
         String image = super.GetImage();
         return CONTINUATION.matcher(image).replaceAll("");
     }
+
+    public static CppCharStream newCppCharStream(Reader dstream) {
+        String source = CharStreamFactory.toString(dstream);
+        JavaccTokenDocument document = new JavaccTokenDocument(source) {
```

Comment:
I feel like this would only help in this specific case, where the implementation is awkward because the Cpp module does not yet use the new ant build. That one makes the array private :/